apps/web/server/routers/viewer/slots.tsx

```diff
@@ -151,16 +151,17 @@ export const slotsRouter = createRouter().query("getSchedule", {
     const startTime =
       input.timeZone === "Etc/GMT"
         ? dayjs.utc(input.startTime)
-        : dayjs.utc(input.startTime).tz(input.timeZone, true);
+        : // The time is already sent in UTC, so mark it UTC by passing true flag and without changing the time. After that convert into the timezone
+          dayjs(input.startTime).utc(true).tz(input.timeZone);
     const endTime =
       input.timeZone === "Etc/GMT"
         ? dayjs.utc(input.endTime)
-        : dayjs.utc(input.endTime).tz(input.timeZone, true);
+        : // The time is already sent in UTC, so mark it UTC by passing true flag and without changing the time. After that convert into the timezone
+          dayjs(input.endTime).utc(true).tz(input.timeZone);
```

Comment:
The issue with this line as far as I'm aware is that `dayjs(input.endTime)` will immediately cast to local time, so if you give a UTC date it will auto-convert to system time. utc(true) will then correctly go into UTC mode but the time conversion has already happened at this point. Not an issue if the system is running in +00:00 but it will be within other timezones.